lib/usb/usb_f107.c

```diff
@@ -60,8 +60,7 @@ static usbd_device *stm32f107_usbd_init(void)
 	OTG_FS_GRSTCTL |= OTG_GRSTCTL_CSRST;
 	while (OTG_FS_GRSTCTL & OTG_GRSTCTL_CSRST);
 
-	uint32_t id = OTG_FS_CID;
-	if (id == OTG_CID_HAS_VBDEN) {
+	if (OTG_FS_CID == OTG_CID_HAS_VBDEN) {
```

Comment:
the body of if statement can do the purpose, and the whole check `CID == HAS_VBDEN` can be removed.
and,
VBDEN and VBUSSEN maps to same bit.
After all, if the SDIS bit is already in clear state (on older uC), cleared on older uC would result in no change.